dbms/src/Interpreters/Aggregator.cpp

```diff
@@ -902,6 +899,8 @@ void Aggregator::spill(AggregatedDataVariants & data_variants)
     data_variants.aggregates_pools = Arenas(1, std::make_shared<Arena>());
     data_variants.aggregates_pool = data_variants.aggregates_pools.back().get();
     data_variants.without_key = nullptr;
+    auto size_bytes = data_variants.bytesCount();
+    agg_spill_context->updatePerThreadRevocableMemory(size_bytes, thread_num);
```

Comment:
Should `updatePerThreadRevocableMemory(0, thread_num)` be called in `tryMarkSpill`?
Because `data_variants` is marked as need spill, this memory can no longer be spilled